Intersect (Core)/Extensions/EnumerableExtensions.cs

```diff
@@ -39,5 +44,77 @@ public static IEnumerable<T> SelectManyRecursive<T>(this IEnumerable<T> source,
 
             return result.Any() ? result.Concat(result.SelectManyRecursive(selector, cancellationToken)) : result;
         }
+
+        public static string GetDescription(this Enum value)
+        {
+            Type type = value.GetType();
+            string name = Enum.GetName(type, value);
+            if (name != null)
+            {
+                FieldInfo field = type.GetField(name);
+                if (field != null)
+                {
+                    DescriptionAttribute attr =
+                           Attribute.GetCustomAttribute(field,
+                             typeof(DescriptionAttribute)) as DescriptionAttribute;
+                    if (attr != null)
+                    {
+                        return attr.Description;
+                    }
+                }
+            }
+            return string.Empty;
+        }
+
+        public static string[] GetDescriptions(Type enumType, string except = null)
+        {
+            return Enum.GetValues(enumType)
+                .Cast<Enum>()
+                .Select(value => value.GetDescription())
+                .Where(value => string.IsNullOrEmpty(except) || value != except)
+                .ToArray();
+        }
+
+        public static GameObjectType GetRelatedTable(this Enum value)
+        {
+            Type type = value.GetType();
+            string name = Enum.GetName(type, value);
+            if (name != null)
+            {
+                FieldInfo field = type.GetField(name);
+                if (field != null)
+                {
+                    RelatedTable attr =
+                           Attribute.GetCustomAttribute(field,
+                             typeof(RelatedTable)) as RelatedTable;
+                    if (attr != null)
+                    {
+                        return attr.TableType;
+                    }
+                }
+            }
+            throw new ArgumentException($"{nameof(value)} did not have a valid RelatedTable attribute to pull from");
+        }
+
+        public static VariableType GetRelatedVariableType(this Enum value)
+        {
+            Type type = value.GetType();
+            string name = Enum.GetName(type, value);
+            if (name != null)
+            {
+                FieldInfo field = type.GetField(name);
+                if (field != null)
+                {
+                    RelatedVariableType attr =
+                           Attribute.GetCustomAttribute(field,
+                             typeof(RelatedVariableType)) as RelatedVariableType;
+                    if (attr != null)
+                    {
+                        return attr.VariableType;
+                    }
+                }
+            }
+            throw new ArgumentException($"{nameof(value)} did not have a valid RelatedVariableType attribute to pull from");
```

Comment:
Invert the if statements.

First check before even grabbing the name should be `Enum.IsDefined(value)` and if it is not, throw saying it's not a valid enum value

Second check is name, throwing saying that the name was empty

Third check is if it found the field info, and if it did not it should throw saying reflection failed

This check order also applies to GetRelatedTable

Reply:
Done